Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/person/ContactDiaryAddPersonViewModel.kt

```diff
@@ -10,46 +10,59 @@ import de.rki.coronawarnapp.contactdiary.storage.entity.ContactDiaryPersonEntity
 import de.rki.coronawarnapp.contactdiary.storage.repo.ContactDiaryRepository
 import de.rki.coronawarnapp.exception.ExceptionCategory
 import de.rki.coronawarnapp.exception.reporting.report
-import de.rki.coronawarnapp.contactdiary.util.formatContactDiaryNameField
-import de.rki.coronawarnapp.ui.SingleLiveEvent
 import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.ui.SingleLiveEvent
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
 import kotlinx.coroutines.CoroutineExceptionHandler
 import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.firstOrNull
 import kotlinx.coroutines.flow.map
 import org.joda.time.LocalDate
 
-class ContactDiaryPersonBottomSheetDialogViewModel @AssistedInject constructor(
+class ContactDiaryAddPersonViewModel @AssistedInject constructor(
     dispatcherProvider: DispatcherProvider,
     @Assisted private val addedAt: String?,
     private val contactDiaryRepository: ContactDiaryRepository
 ) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
     private val coroutineExceptionHandler = CoroutineExceptionHandler { _, ex ->
-        shouldClose.postValue(null)
         ex.report(ExceptionCategory.INTERNAL, TAG)
     }
 
-    private val text = MutableStateFlow("")
+    val shouldClose = SingleLiveEvent<Unit>()
 
-    val isValid = text.map {
-        it.isNotEmpty() && it.length <= MAX_PERSON_NAME_LENGTH
-    }.asLiveData()
+    private val _name = MutableStateFlow("")
+    val name: StateFlow<String> = _name
 
-    val shouldClose = SingleLiveEvent<Unit>()
+    private val _phoneNumber = MutableStateFlow("")
+    val phoneNumber: StateFlow<String> = _phoneNumber
+
+    private val _emailAddress = MutableStateFlow("")
+    val emailAddress: StateFlow<String> = _emailAddress
+
+    val isNameValid = name
+        .map { it.isNotEmpty() }
+        .asLiveData()
 
-    private val formattedName: String
-        get() = text.value.formatContactDiaryNameField(MAX_PERSON_NAME_LENGTH)
+    fun nameChanged(value: String) {
+        _name.value = value.trim().take(MAX_PERSON_NAME_LENGTH)
```

Comment:
The PR description says that we should check for all fields if not more than 250 characters can be entered. But you are only taking 250 characters for the name, and not phone number or email adress? Same for location.

Reply:
I'm thinking of removing this check completely since it's kind of redundant as I'm limiting the text fields to 250 characters from the xml.